pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -160,6 +175,78 @@
 }
 
 
+@dataclass
+class _OpenAIToolChoiceResult:
```

Comment:
I think having 2 classes for intermediate representations of tool choice is too much; I'd rather have the logic inline like we do in the other model classes.